factory/processing/processComponents_test.go

```diff
@@ -655,6 +659,67 @@ func TestNewProcessComponentsFactory(t *testing.T) {
 		require.NotNil(t, pcf)
 	})
 }
+func TestProcessComponentsFactory_AddSystemVMToContainer(t *testing.T) {
+	t.Parallel()
+
+	testContainerKey := "testContainer"
+
+	t.Run("sovereign should add systemVM", func(t *testing.T) {
+		t.Parallel()
+
+		shardCoordinator := mock.NewMultiShardsCoordinatorMock(2)
+		processArgs := components.GetProcessComponentsFactoryArgs(shardCoordinator)
+		processArgs.ChainRunType = common.ChainRunTypeSovereign
+		pcf, _ := processComp.NewProcessComponentsFactory(processArgs)
+		require.NotNil(t, pcf)
+		vms := make(map[string]vmcommon.VMExecutionHandler, 0)
+		vms[testContainerKey] = &processMock.VMExecutionHandlerStub{}
+		vmContainer := &processMock.VMContainerMock{
+			AddCalled: func(key []byte, val vmcommon.VMExecutionHandler) error {
+				vms[string(key)] = val
+				return nil
+			},
+		}
+		builtInFunctions := &mock.BuiltInFunctionFactoryMock{}
+
+		err := pcf.AddSystemVMToContainerIfNeeded(vmContainer, builtInFunctions)
+
+		require.NoError(t, err)
+		require.NotNil(t, vmContainer)
+
+		require.Equal(t, 2, len(vms))
+		require.NotNil(t, vms[string(vmFactory.SystemVirtualMachine)])
+		require.NotNil(t, vms[testContainerKey])
+		require.Equal(t, "*process.systemVM", fmt.Sprintf("%T", vms[string(vmFactory.SystemVirtualMachine)]))
+	})
+	t.Run("shard should NOT add systemVM", func(t *testing.T) {
+		t.Parallel()
+
+		shardCoordinator := mock.NewMultiShardsCoordinatorMock(2)
+		processArgs := components.GetProcessComponentsFactoryArgs(shardCoordinator)
+		processArgs.ChainRunType = common.ChainRunTypeRegular
+		pcf, _ := processComp.NewProcessComponentsFactory(processArgs)
+		require.NotNil(t, pcf)
+		vms := make(map[string]vmcommon.VMExecutionHandler, 0)
```

Comment:
```suggestion
		require.NotNil(t, pcf)
		
		vms := make(map[string]vmcommon.VMExecutionHandler, 0)
```

Same suggestion for the above test

Reply:
done